packages/device/src/classes/device.ts

```diff
@@ -214,8 +222,35 @@ export class Device extends EventEmitter {
     this.player = undefined;
   }
 
+  /**
+   * Get the credentials from the integration and validate them.
+   * If the credentials are valid, return them.
+   */
+  async getCredentials(): Promise<Credentials | undefined> {
+    const credentials = await this.integration.getCredentials();
```

Comment:
Hmm. Yes error handling is probably pretty much ignored here and in other places where we talk to the integration.
The integration should either return a credentials json-object or undefined if it has no stored credentials. However, if something goes wrong in the integration code and an exception is thrown, what do we do? I guess this boils down to how we want to handle misbehaving integrations. One possibility would be to wrap all integration calls in a try catch and log any errors thrown to logger. Another possibility would be to let it crash and have some generic top level error handling in place.

Reply:
I gave this some thought and I think we should just trust that the integration follows the API here. If the integration thows an error here something fatal happened and nothing will work anyway. This could happen for sync apis as well by the way. A high level error handling that reports such (and other uncatched) errors would of course make sense though...

If there was some case where a rejected promise was the expected result, the situation would be different but here we should get either a string if the credentials exist or undefined if the don't. No other expected cases exist.